### DISCUSSION TITLE
Custom Query => no checkboxes in table

### DISCUSSION THREAD

**Discussion by VenSlazer** [Budibase/budibase#11152] [Help] [answered]
Hey there,

I have switched to custom queries due to performance issues. Budibase runs much faster, but what I noticed is, that on my tables, boolean values are now shown as "true" or "false", instead of showing a checkbox. How can I get those checkboxes back?

<img width="1338" alt="Bildschirmfoto 2023-07-06 um 14 18 38" src="https://github.com/Budibase/budibase/assets/77432946/2b84865c-25be-44ea-b4b8-ea2b2448c73f">

**Comment by R2bEEaton** [comment, marked as answer]
Hi @VenSlazer, when using custom queries sometimes Budibase doesn't correctly identify the types of each column.

So at the bottom of your custom query, choose the Schema view:

![image](https://github.com/Budibase/budibase/assets/34921506/81692e3b-5651-46c1-b78b-15aca1f83f80)
![image](https://github.com/Budibase/budibase/assets/34921506/17d72bbd-d685-44af-b51c-a0691f3d9dea)

As you can see, my "Read" column (which is a boolean in MS SQL) was detected as "Text" automatically. However, I can use the dropdown list to correct it. After changing it to "Boolean" my table now looks like this, with the checkboxes returned.

![image](https://github.com/Budibase/budibase/assets/34921506/f7fead84-aec0-456d-9e57-719bcddee19b)

### Hope this helps!
Let me know if you have any further questions. Otherwise, make sure to mark this question as answered to help future readers. 😃

**P.S.** Table columns can be customized pretty heavily. For example, one possible solution would be to use [advanced column customization](https://docs.budibase.com/docs/table#advanced-column-customization) to display a checkbox emoji ☑️ when `true` and an empty square when `false` using Handlebars. Pretty powerful stuff, but obviously unnecessary in this case. `{{ #eq Value true }}☑{{ else }}☐{{ /eq }}`

**Reply by VenSlazer** [reply]
Hey @R2bEEaton , 

Thank you very much for this very detailed answer. Awesome! => That solved my Problem. Thank you!

**Reply by R2bEEaton** [reply]
Glad to hear! You're welcome! Definitely go ahead and click the checkbox next to my answer to mark the question as answered. (You closed the question, which is different.)

**Reply by VenSlazer** [reply]
Done ;-)
